### ISSUE TITLE
Minimap

### ISSUE THREAD

**Issue by rocketinventor** [ajaxorg/ace#2655]
Hi, I've been hard at work on a minimap plugin for C9/ace.
The url is https://github.com/rocketinventor/ace-minimap.
However, there are some parts that I still haven't gotten/ cannot get worked out and there are a few parts it needs to be functional, so please check https://github.com/rocketinventor/ace-minimap/issues to see how you can help.
Please do clone, fork, star, commit, and pull request.
Thanks,
-Elliot

**Comment by mlajtos** [comment]
Hey! Is there any demo available?

**Comment by rocketinventor** [comment]
No, sorry. I haven't gotten it to do anything useful, so no demo available.
I would like to get a functional build out asap but, I am having problems getting certain things to work.
You can git clone to the plugins repository to try out the current work, though.

**Comment by OmlineEditor** [comment]
I can't install the plugin
![map](https://cloud.githubusercontent.com/assets/13348933/9812348/7023c3f2-5885-11e5-96db-9440506bea9c.png)

**Comment by rocketinventor** [comment]
@OmlineEditor Why are you trying to upload it?

**Comment by OmlineEditor** [comment]
I add a package from the list (you can check by command: c9 list).
after I wanted to install command: c9 install "Ace Minimap". 
what am I doing wrong? how to install other plugins. please help

**Comment by rocketinventor** [comment]
Oh, I don't know about the list. I never added ace-minimap to it so I wouldn't suggest trying to install it that way yet. Neither ace's plugin system, nor this plugin are quite ready for primetime yet.
Sorry (all help is welcome, though).

**Comment by OmlineEditor** [comment]
do not delete

**Comment by reactjser** [comment]
How is it going?

**Comment by ldijkman** [comment]
Ace minimap

https://github.com/ldijkman/Ace_MiniMap/tree/main/docs

![Uploading 269827684-0b4c9f3a-6d8d-4879-ab6d-c148f4b624fb.png…]()

**Comment by marinsokol5** [comment]
Nice work!
We do have Ace extension folder, in case you want to share it with others https://github.com/ajaxorg/ace/tree/master/src/ext by opening us a PR.

@nightwing is this something that we would be interesting from a feature perspective?

**Comment by whazor** [comment]
Also, it would be better to use [static_highlight.js](src/ext/static_highlight.js) instead of a full-fledged ace-editor because the minimap does not need any editing features, so static highlighting would be more optimal.

**Comment by ldijkman** [comment]
made a few small changes
not perfect, but works a bit better

https://ldijkman.github.io/Ace_MiniMap/ace_editor_minimap.html

![2023-09-22-220255_1920x1080_scrot](https://github.com/ajaxorg/ace/assets/45427770/8b3db1fc-a70a-42c5-b7d5-1039e0dd0088)

**Comment by andredcoliveira** [comment]
This is a very useful contribution! I think a lot of people expect this feature from an editor nowadays.

Would you be able to open a PR on this repository, @ldijkman, as @marinsokol5 suggested above?

Also, a few notes after playing around with https://github.com/ldijkman/Ace_MiniMap/tree/main/docs, mostly from a UI/UX perspective:
- positioning the minimap immediately adjacent to the main editor's scrollbar could economize horizontal space
  - currently, there is some space between the minimap and the scrollbar that I imagine isn't effective to use
- I'm a bit torn on being able to interact with the text on the minimap, probably because of familarity with VSCode's minimap.
  - I would suggest, if possible, to turn the minimap into one big scrollbar, though I'm not sure if Ace allows for that at this moment
- positioning the cursor on the minimap places it in the same position in the main editor, but the opposite doesn't seem to be true
  - positioning the cursor in the main editor doesn't results in offset lines being selected in the minimap

**Comment by ldijkman** [comment]
Think this one ads an minimap to the editor 

https://ldijkman.github.io/Ace_Seventh_Heaven/Hell.html

**Comment by TimChinye** [comment]
> Also, it would be better to use [static_highlight.js](https://github.com/ajaxorg/ace/tree/master/src/ext/static_highlight.js) instead of a full-fledged ace-editor because the minimap does not need any editing features, so static highlighting would be more optimal.

Highly agree with this! Plus, let’s talk about the minimap’s look—it’s just not doing it for me. Sure, you can tweak it with some CSS magic, but the out-of-the-box style isn’t exactly inviting.

Also, there’s this quirky thing in the demo where the cursor’s line marker on the minimap snaps to where you click, but then it hops up a line right away. Definitely a bug, but honestly, it’s one we could do without if the highlight just stayed put.

With the limited availability for solid minimap extensions, I’m skipping the minimap feature for my project’s first version. When it’s time for updates, I’ll circle back to see if the "Ace Minimap" has stepped up its game. If not, it might be DIY time for me.
